common/src/main/java/com/wynntils/models/gearinfo/GearInfoRegistry.java

```diff
@@ -0,0 +1,420 @@
+import com.wynntils.models.gearinfo.type.GearDropType;
+import com.wynntils.models.gearinfo.type.GearInfo;
+import com.wynntils.models.gearinfo.type.GearMajorId;
+import com.wynntils.models.gearinfo.type.GearMaterial;
+import com.wynntils.models.gearinfo.type.GearMetaInfo;
+import com.wynntils.models.gearinfo.type.GearRequirements;
+import com.wynntils.models.gearinfo.type.GearRestrictions;
+import com.wynntils.models.gearinfo.type.GearTier;
+import com.wynntils.models.gearinfo.type.GearType;
+import com.wynntils.models.stats.FixedStats;
+import com.wynntils.models.stats.StatCalculator;
+import com.wynntils.models.stats.type.DamageType;
+import com.wynntils.models.stats.type.StatPossibleValues;
+import com.wynntils.models.stats.type.StatType;
+import com.wynntils.utils.JsonUtils;
+import com.wynntils.utils.colors.CustomColor;
+import com.wynntils.utils.type.Pair;
+import com.wynntils.utils.type.RangedValue;
+import com.wynntils.utils.wynn.WynnUtils;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+
+public class GearInfoRegistry {
+    private List<GearMajorId> allMajorIds = List.of();
+    List<GearInfo> gearInfoRegistry = List.of();
+    Map<String, GearInfo> gearInfoLookup = Map.of();
+    Map<String, GearInfo> gearInfoLookupApiName = Map.of();
+
+    public GearInfoRegistry() {
+        loadRegistry();
+    }
+
+    public void reloadData() {
+        loadRegistry();
+    }
+
+    private void loadRegistry() {
+        // We must download and parse Major IDs before attempting to parse the gear DB
+        Download majorIdsDl = Managers.Net.download(UrlId.DATA_STATIC_MAJOR_IDS);
+        majorIdsDl.handleReader(majorIdsReader -> {
+            Type type = new TypeToken<List<GearMajorId>>() {}.getType();
+            Gson majorIdGson = new GsonBuilder()
+                    .registerTypeHierarchyAdapter(GearMajorId.class, new GearMajorIdDeserializer())
+                    .create();
+            allMajorIds = majorIdGson.fromJson(majorIdsReader, type);
+
+            // Now we can do the gear DB
+            Download dl = Managers.Net.download(UrlId.DATA_STATIC_GEAR);
+            dl.handleReader(reader -> {
+                Gson gearInfoGson = new GsonBuilder()
+                        .registerTypeHierarchyAdapter(GearInfo.class, new GearInfoDeserializer(allMajorIds))
+                        .create();
+                WynncraftGearInfoResponse gearInfoResponse =
+                        gearInfoGson.fromJson(reader, WynncraftGearInfoResponse.class);
+
+                // Remove the dummy "default" entry
+                List<GearInfo> registry = gearInfoResponse.items.stream()
+                        .filter(gearInfo -> !gearInfo.name().equals("default"))
+                        .toList();
```

Comment:
What do we filter here? Do we remove a singular item from the API?

Reply:
Yes. This was the very last item as returned in the API, with I believe all stats at 0 and the lore text "This is a rad weapon yo. Let's make the lore longer so we can test how long it can go. Hi Jumla!". It did not show up in Wynnbuilder or Wynndata. So I decided to filter it out. 

Then I was told that there are more test items in the data set, so we should probably expand this filtering to check an entire "blocklist" of items that are not for real, and put this in a json file, so we can update it easier. 

I forgot about that, though. My suggestion is that we have just the one filtering of "default" right now, and write a more general filtering mechanism as a follow-up.